server/src/test/java/org/cloudfoundry/identity/uaa/oauth/UaaTokenStoreTests.java

```diff
@@ -450,5 +468,24 @@ public Object invoke(Object proxy, Method method, Object[] args) throws Throwabl
         }
     }
 
+    public class TestClock extends Clock {
+        public Queue<Instant> instants = new LinkedList<Instant>();
+
+        @Override
+        public ZoneId getZone() {
+            return ZoneId.systemDefault();
+        }
+
+        @Override
+        public Clock withZone(ZoneId zone) {
+            return this;
+        }
+
+        @Override
+        public Instant instant() {
+            return instants.remove();
+        }
+    }
+
```

Comment:
That's the only way I found to do what I wanted to do. Please advice if you know a better way.

Reply:
my advise is , use mockito for the expiration, see https://github.com/cloudfoundry/uaa/pull/2292